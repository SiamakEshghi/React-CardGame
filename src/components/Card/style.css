.card{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-around;
    height: 200px;
    width: 150px;
    background-color: white;
    border-radius: 5px;
}
.icon {
    display: flex;
    align-self: 'left';
    max-height: 30px;
}
.back {
    max-height: 200px;
    width: auto;
}