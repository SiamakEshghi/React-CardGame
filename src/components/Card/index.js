import React, { Fragment } from 'react';
import ReactLoading from 'react-loading';
import 'components/Card/style.css';
import { getIcon } from 'utils';
import cardBack from 'img/cardBack.png';

export default ({ suite = "", value = "", back, loading }) => {

    function renderCard () {
        if(loading) {
            return(
                <ReactLoading type="cylon" color="black"/>
            );
        }
        if(back) {
            return(
                <img className="back" src={cardBack}/>
            );
        }
        if(!suite || !value) {
            return(
                <h4>No More Card!</h4>
            );
        }
        return (
            <Fragment>
                <img className="icon" src={getIcon(suite)}/>
                <h3>{value}</h3>
            </Fragment>
        );

    }
    return (
        <div className="card">
            {renderCard()}
        </div>
    );
}