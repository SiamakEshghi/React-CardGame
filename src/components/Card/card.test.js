import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Card from 'components/Card';

let wrapper;

beforeEach(() => {
    wrapper = shallow(<Card />);
});

describe('Card', () => {
    it('Should render correctly', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('If value is empty just show No More Card!', () => {
        wrapper.setProps({value : ""});
        wrapper.update();
        expect(wrapper.find('.back').length).toEqual(0);
        expect(wrapper.find('Loading').length).toEqual(0);
        expect(wrapper.render().text()).toContain('No More Card!');
    });

    it('If back is true just show back card', () => {
        wrapper.setProps({back: true});
        wrapper.update();
        expect(wrapper.find('.back').length).toEqual(1);
        expect(wrapper.find('Loading').length).toEqual(0);
        expect(wrapper.render().text()).not.toContain('No More Card!');
    });
    it('If loading is true just show loading', () => {
        wrapper.setProps({loading: true});
        wrapper.update();
        expect(wrapper.find('.back').length).toEqual(0);
        expect(wrapper.find('Loading').length).toEqual(1);
        expect(wrapper.render().text()).not.toContain('No More Card!');
    });
});