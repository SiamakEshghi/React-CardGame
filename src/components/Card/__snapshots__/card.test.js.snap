// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should render correctly 1`] = `
<div
  className="card"
>
  <h4>
    No More Card!
  </h4>
</div>
`;
